### PR TITLE
Add GET k8sd/node endpoint

### PR DESCRIPTION
**Pull request by bschimke95**
Provides an endpoint to query the current status of the local k8s node. This is especially useful for the charm that wants to know the current role of specific node. It is not possible to query other nodes for security purposes.

Additionally,
* changed `ClusterMember` to `NodeStatus` to better reflect the new shared use
* added a hidden CLI command `k8s local-node-status` for testing.
* added basic test coverage by integrating assertions into the `test_clustering.py` test.
* use enum for datastore and cluster roles to not rely on random strings.

### CODE REVIEW

File: src/k8s/cmd/k8s/k8s_local_node_status.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+			if err != nil {
+				return fmt.Errorf("failed to get cluster status: %w", err)
+			}
+
+			fmt.Println(clusterStatus)
+			return nil
+		},
+	}
+	return localNodeStatusCmd
+}
```
Comment (mateoflorido) [inline src/k8s/cmd/k8s/k8s_local_node_status.go:29]: Are we keeping this as a hidden command? What if we could repurpose it and make it a hidden subcommand of the k8s status? If it does not deliver relevant information, I'm OK to just dropping it.

File: src/k8s/pkg/k8sd/api/impl/k8sd.go [hunk 2]
```diff
@@ -58,21 +60,50 @@ func GetClusterMembers(ctx context.Context, s *state.State) ([]apiv1.ClusterMemb
 		return nil, fmt.Errorf("failed to get cluster members: %w", err)
 	}
 
-	members := make([]apiv1.ClusterMember, len(clusterMembers))
+	members := make([]apiv1.NodeStatus, len(clusterMembers))
 	for i, clusterMember := range clusterMembers {
-		fingerprint, err := shared.CertFingerprintStr(clusterMember.Certificate.String())
```
Comment (mateoflorido) [inline src/k8s/pkg/k8sd/api/impl/k8sd.go:63]: Is `fingerprint` no longer required?

**Review by mateoflorido — APPROVED** [review]
Just a couple of questions out of curiosity. LGTM!

File: src/k8s/api/v1/types.go [hunk 3]
```diff
@@ -48,28 +48,54 @@ func BootstrapConfigFromMap(m map[string]string) (*BootstrapConfig, error) {
...
+
+// DatastoreRole as provided by dqlite
+type DatastoreRole string
+
+const (
+	DatastoreRoleVoter   DatastoreRole = "voter"
+	DatastoreRoleStandBy DatastoreRole = "stand-by"
+	DatastoreRoleSpare   DatastoreRole = "spare"
+	DatastoreRolePending DatastoreRole = "PENDING"
+	DatastoreRoleUnknown DatastoreRole = "unknown"
```
[the remark this answers is not in the document]
Comment (bschimke95) [reply src/k8s/api/v1/types.go:69]: Super weird, isn't it?
Complains directly to the dqlite team please ;)

File: src/k8s/cmd/k8s/k8s_local_node_status.go (hunk 1 above)
Comment (bschimke95) [reply src/k8s/cmd/k8s/k8s_local_node_status.go:29]: This is really just a hidden command for testing and debugging purpose. We could drop it but I also don't see a problem with keeping it.

File: src/k8s/pkg/k8sd/api/impl/k8sd.go (hunk 2 above)
Comment (bschimke95) [reply src/k8s/pkg/k8sd/api/impl/k8sd.go:63]: It was never. Just a leftover from early days where I randomly copied Microcluster examples.

File: src/k8s/api/v1/types.go [hunk 4]
```diff
@@ -48,28 +48,54 @@ func BootstrapConfigFromMap(m map[string]string) (*BootstrapConfig, error) {
...
+)
+
+// NodeStatus holds information about a node in the k8s cluster.
+type NodeStatus struct {
+	// Name is the name for this cluster member that was when joining the cluster.
+	// This is typically the hostname of the node.
+	Name string `mapstructure:"name,omitempty"`
```
Comment (neoaggelos) [inline src/k8s/api/v1/types.go:76]: what are the mapstructure tags for? let's just use `json`.

**Review by neoaggelos — APPROVED** [review]
LGTM, just one comment

**Pull request merged**